inference/core/workflows/core_steps/models/third_party/qr_code_detection.py

```diff
@@ -71,62 +68,57 @@ def get_manifest(cls) -> Type[WorkflowBlockManifest]:
-            detections["data"] = np.array(
-                [p.get("data", "") for p in prediction["predictions"]]
-            )
-            prediction["predictions"] = detections
-        converted_predictions = attach_prediction_type_info(
-            predictions=predictions,
-            prediction_type="qrcode-detection",
-        )
-        converted_predictions = attach_parent_info(
-            images=image, predictions=converted_predictions
-        )
-        return anchor_prediction_detections_in_parent_coordinates(
-            image=image,
-            predictions=converted_predictions,
+        results = []
+        for image in images.iter_nonempty():
+            qr_code_detections = detect_qr_codes(image=image)
+            results.append({"predictions": qr_code_detections})
+        return images.align_batch_results(
+            results=results, null_element={"predictions": None}
         )
 
 
-def detect_qr_codes(image: np.ndarray) -> List[dict]:
+def detect_qr_codes(image: WorkflowImageData) -> sv.Detections:
     detector = cv2.QRCodeDetector()
-    retval, detections, points_list, _ = detector.detectAndDecodeMulti(image)
-    predictions = []
+    retval, detections, points_list, _ = detector.detectAndDecodeMulti(
+        image.numpy_image
+    )
+    xyxy = []
+    confidence = []
+    class_id = []
+    class_name = []
+    extracted_data = []
     for data, points in zip(detections, points_list):
         width = points[2][0] - points[0][0]
         height = points[2][1] - points[0][1]
-        predictions.append(
-            {
-                "class": "qr_code",
-                "class_id": 0,
-                "confidence": 1.0,
-                "x": points[0][0] + width / 2,
-                "y": points[0][1] + height / 2,
-                "width": width,
-                "height": height,
-                "detection_id": str(uuid4()),
-                "data": data,
-            }
-        )
-    return predictions
+        x_min = points[0][0]
+        y_min = points[0][1]
+        x_max = x_min + width
+        y_max = y_min + height
+        xyxy.append([x_min, y_min, x_max, y_max])
+        class_id.append(0)
+        class_name.append("qr_code")
+        confidence.append(1.0)
+        extracted_data.append(data)
+    xyxy = np.array(xyxy) if len(xyxy) > 0 else np.empty((0, 4))
+    confidence = np.array(confidence) if len(confidence) > 0 else np.empty(0)
+    class_id = np.array(class_id).astype(int) if len(class_id) > 0 else np.empty(0)
+    class_name = np.array(class_name) if len(class_name) > 0 else np.empty(0)
+    detections = sv.Detections(
+        xyxy=np.array(xyxy),
+        confidence=confidence,
+        class_id=class_id,
+        data={CLASS_NAME_DATA_FIELD: class_name},
+    )
+    detections[DETECTION_ID_KEY] = np.array([uuid4() for _ in range(len(detections))])
+    detections[PREDICTION_TYPE_KEY] = np.array(
+        ["qrcode-detection" for _ in range(len(detections))]
+    )
```

Comment:
Loop can be avoided: `detections[PREDICTION_TYPE_KEY] = np.array(["qrcode-detection"] * len(detections))`

Reply:
👍